qmcpy/algorithms/accum_data/MeanVarData.py

```diff
@@ -23,7 +23,7 @@ def __init__(self, nf: int) -> None:
         self.nSigma = zeros(nf) # number of samples used to compute the sample standard deviation
         self.nMu = zeros(nf)  # number of samples used to compute the sample mean
 
-    def update_data(self, distrib_obj: DiscreteDistribution, fun_obj: Fun) -> None:
+    def update_data(self, distrib_obj: DiscreteDistribution, fun_obj: IntegrandBase) -> None:
```

Comment:
I'm not someone who likes typing in Python, but that's a personal preference.  Facebook is all about it now, Google seems ambivalent, Uber doesn't do it ... opinions across the spectrum.  I would say, though, that if you want to have these type suggestions that it should be done consistently throughout.  Nothing wrong with starting one place and continuing, but it should be decided whether this is desired or not.

Reply:
we've been adding typing to files as they get updated documentation.
I'm not sure if someone who wants to contribute to our software would be familiar and ready to use typing.
Maybe it is a good idea in the abstract classes but not in the concrete implementations? Would this break consistency? Just typing our softwares objects?

Reply:
That's part of the reason I don't generally bother doing it at all -- it's a lot to keep up throughout the code base for, what is often, almost no real benefit.

Given that, one advantage of limiting merge control is that before you let external contributors push content to master they have to satisfy your expectations.  If an expectation is that types are defined, then you can enforce that and hold up a PR until it's fixed.